test/consumer.test.js

```diff
@@ -19,28 +18,42 @@ describe('Consumer', () => {
   before(() => {
     Nsqd = lib.api.Nsqd;
     nsqd = new Nsqd(nsqdHTTPAddress);
+    Consumer = lib.Consumer;
   });
 
   it('should be there', () => {
     lib.should.have.property('Consumer').which.is.Function();
-    Consumer = lib.Consumer;
   });
 
-  it('should receive message successfully', (done) => {
+  it('should receive message successfully', done => {
     const topic = randexp(/Consume-([a-z]{8})/);
     nsqd.publish(topic, 'hello nsq').then(() => {
       const c = new Consumer(topic, 'ipsum', {
         lookupdHTTPAddresses
       });
-      c.consume((msg) => {
+      c.consume(msg => {
         expect(msg.body.toString()).to.be.equal('hello nsq');
         msg.finish();
         removeTopicFromAllNsqd(topic, done);
       });
     }, done);
   });
 
-  it('should throw error if connect after auto connection', (done) => {
+  it('should receive message successfully', done => {
+    const topic = randexp(/Consume-([a-z]{8})/);
+    nsqd.deferPublish(topic, 'hello delay', 10000).then(() => {
```

Comment:
shouldn't it be tested against publisher since you extended it with `deferPublish`, there is no point to test `nsqd`.

Reply:
this is not for testing nsqd, but to test consumer can successfully receive a delay msg.

Reply:
You changed the publisher(by calling deferPublish) but why you test the consumer(or the NSQ function)?

Reply:
As we added delay feature, we need to make sure:
- Producer can push a delay msg successfully
- Consumer can receive a delay msg successfully(which is what you see here)

Why do you think there's no need to test consumer? which is not a NSQ function but defined by the lib.

Reply:
I guess you should just move the test to an end-to-end test or a use-case test

Reply:
I was actually copying the pattern of the [test](https://github.com/Wiredcraft/nsq-strategies/blob/master/test/consumer.test.js#L70-L84) you wrote for Consumer in consumer.test.js. If consumer.test.js is no more needed, feel free to move or remove it.